src/data/feature_index.js

```diff
@@ -111,6 +119,8 @@ class FeatureIndex {
         // expanded by to include these features.
         let additionalRadius = 0;
         for (const id in styleLayers) {
+            if (!this.knownLayers[id]) continue;
```

Comment:
Is this code especially hot? Could we get by with a less stateful & slightly less efficient `this.hasLayer(id)` method?